modules/flashprog

```diff
@@ -0,0 +1,46 @@
+modules-$(CONFIG_FLASHPROG) += flashprog
+
+flashprog_depends := pciutils $(musl_dep)
+
+flashprog_version := 9dc6d843b0678001c9baf0e8602ecb25b16329d2
+flashprog_dir := flashprog-$(flashprog_version)
+flashprog_tar := $(flashprog_dir).tar.gz
+flashprog_url := https://github.com/SourceArcade/flashprog/archive/$(flashprog_version).tar.gz
+flashprog_hash := fa4ddf3b60314994a37e4599122ae4c7f42135c13c782e580bc580d715cfa2fc
+
+# Default options for flashprog
+flashprog_cfg := \
+	WARNERROR=no \
+	CONFIG_NOTHING=yes \
+	CONFIG_INTERNAL=yes \
+	CONFIG_INTERNAL_X86=yes \
+
+ifeq "$(CONFIG_TARGET_ARCH)" "ppc64"
+flashprog_cfg := \
+	WARNERROR=no \
+	CONFIG_NOTHING=yes \
+	CONFIG_LINUX_MTD=yes
+endif
+
+#Only enable AST1100 if requested per board configs
+ifeq "$(CONFIG_FLASHPROG_AST1100)" "y"
```

Comment:
Verify

Reply:
Patch missing. Even i currently activated under server board and passed at make, doesn't o anything without https://github.com/linuxboot/heads/blob/master/patches/flashrom-b1f858f65b2abd276542650d8cb9e382da258967/0100-enable-kgpe-d16.patch missing

@i-c-o-n?

Reply:
I can have a look at it. AFAIR, nobody made a real effort to upstream this. If I rebase this, we'd still need somebody to test it thoroughly. We are quite pedantic when it comes to internal flashing.

Reply:
@i-c-o-n it is a problem of upstreaming indeed, keeping track of patches downstream never proved to do any good.

If not under flashprog/flashrom: this means end users need to flash bmc externally which is not a desirable thing. Also as you may know, that bmc port is old but still in charge of best fan control out there for the d16. Having heads flash through https://github.com/linuxboot/heads/pull/1769/files#diff-08a8a9080a4ca4377e0de4dbef6bd94c84209343b4e12a8982352a1c1b09b71b would be useful.

That would be ideal, but not a blocker for this PR.

Reply:
@i-c-o-n do you see anything else that should be improved in the flashprog makefile for size/functionalities inclusion?

Reply:
Do you see anything that should be added in board config for flash options that should be passed? Something that would make speed of internal flashing faster?

Also not a requirement for this PR, but a nice thing to track in separate issue for improvements, let me know.

Reply:
> @i-c-o-n do you see anything else that should be improved in the flashprog makefile for size/functionalities inclusion?

No, the current selection is already the minimum AFAICT. Anything else would need some hacking on the code. One low hanging fruit: For targets that use Intel hwseq or linux_mtd, you could drop everything else from `flashchips.c`. Haven't tried, but it probably won't link the chip drivers then.

> Do you see anything that should be added in board config for flash options that should be passed? Something that would make speed of internal flashing faster?

`--noverify-all` gives you a little reading-speed advantage. But assuming that you update most of the flash anyway, it won't make much of a difference.